### PR TITLE
Consensus seeking edits for harassment policy

### PR DESCRIPTION
**Pull request by KelseyDH**
Reading some the discussion [here.](https://github.com/MrStonedOne/opencodeofconduct/pull/1)  I put together this pull request to try to seek out a consensus of how the Code of Conduct's wording could be improved.

Two useful sources I relied on for research were the Canadian [Human Rights Commission's guide on Harassment](http://www.tbs-sct.gc.ca/psm-fpfm/healthy-sain/prh/gahrp-gaprh-eng.asp#toc3-2-2-1) and [recommendations from a civil liberties organization on how to balance free speech and protecting people from genuine harassment.](https://bccla.org/our_work/comments-on-the-proposed-simon-fraser-university-harassment-policy/)

**Summary of changes:**

**_Concretely defined a test for harassment.**_  
The harassment wording right now is extremely broadly defined.  Nearly any activity of expression could fall under it.  The "harassment" test I added here posits that unless you can make a cogent case that you are _directly_ being targeted by the conduct in question, then you are not being harassed.

**_Allow `unwelcomed comments` to be okay when there is a bona fide justification.**_
This sentence struck me as extremely broad.  Unjustified targeting of ones lifesyle is no doubt discriminatory, but not always.  Asking about somebody's sleep schedule because they are late for work may be _unwelcomed_, but it's _not_ harassment when there is a _bona fide_ justification to bring it up.

**_Remove disputed wording; replace with language protecting activities of disadvantaged groups.**_
This consensus-seeking move preserves the good intentions of what the authors of the original sentence were going for (that their priority is the protection of marginalized people), without necessarily "taking sides" or incorporating language that only narrowly focuses on the priorities of specific rights movements.

**_Prohibit vexatious and bad faith complaints.**_
I strongly recommend the addition of language prohibiting the abuse of the complaints process.  Without it, we are giving free permission to users with axes to grind to go on witch hunts.

---

I welcome any comments, contributions or modifications people may think are appropriate to the changes I'm proposing.  I'm not going to say that I think the current policy is fixed by any means, but at the very least this could make some progress towards creating a better Code of Conduct for all.

### CODE REVIEW

**Comment by ghost** [comment]
Thank you for listening. The revised policy is much clearer in it's intent.

**Comment by An-Annoymous-Developer** [comment]
I like this and I have a couple of suggestions. 

When you have your lists of protected characteristics the law includes [philosophical beliefs](http://www.acas.org.uk/religionorbelief) and there is a movement to include [subcultures](http://www.theguardian.com/uk/2013/apr/03/manchester-police-goths-punks-hate-crime) - perhaps those should be included too. 

The line `We will not tolerate discrimination based on any of the protected characteristics above, including participants with disabilities.` also scans oddly to me. I think it would be better if disabilities (perhaps explicitly mentioning both physical and mental) were included in the list immediately preceding the quoted section.

**Comment by LimeBlast** [comment]
:+1:

**Comment by KelseyDH** [comment]
@An-Annoymous-Developer Thanks for the kind words.  Two points:
1. The sentence mentioning disabilities at the end also caught my attention for being grammatically awkward.  I agree it should be changed.  I didn't fix it as I wanted to reduce the amount of changes I presented in this pull request.
2.  I agree that protecting philosophical beliefs as it relates to spiritual and religious protection sounds like a good move.  However, for protecting subcultures more generally I'm less sure, since subcultures are often proxies for political conflicts over values.  (e.g. think ISIS, hate groups, anti-vaxxers, people who 'doxx' those they don't like).  Either way it's an interesting discussion I'm interested in exploring further.

---

**Note about Harassment vs. Taking "Offense"**

As a personal aside I wanted to clarify that when constructing a harassment policy, I strongly recommend that  we focus only on using the term harassment for actions that are _genuinely harassing to one's ability to participate as equals within an open source community._  Harassment should not be used as a term for describing things that are offensive to us which do not actually target us _directly._

For example:

**Activity that may be offensive, but which _is not_ harassment:**
 Writing in a code repository that you are against abortion.  Or creating a repository on Github that's focused on anti-abortion advocacy.

**Activity that may be offensive which clearly _is_ harassment:**
Identifying or "doxxing" the identify of a woman coder who has had an abortion, and publicly posting in her code repositories that she is "baby killer."  Or threatening to let her priest or parents know about this secret.  (Yes, this kind of harassment _has_ happened elsewhere.)

Without taking a stance on the controversial issue of abortion (where I know opinions vary), I simply bring this up to show how even on a contentious issue where opinions can cause offense, most _reasonable observers_ will agree that the second example I provided is a _genuine issue of harassment_, not just a case of _taking offense_ to the actions of someone else.  My pull request successfully delineates between these two types of behaviours.

**Comment by preoctopus** [comment]
I'm out of my element here, but I got sucked into this when I read that weird "reversisms" line. I'm asking for clarification in another issue so not to discuss that. Just a warning that I'm not well versed in social issues, and if I'm missing an important detail please clarify for me. 

This is nothing more than my $0.02

> Allow unwelcomed comments to be okay when there is a bona fide justification.

This seems rife for abuse in the opposite direction.

Would a clause that gives an opportunity to have _why_ this is harassment be explained make sense? and an opportunity to make amends/explain? aka "Never attribute to malice what can equally be explained by ignorance." 

A: "Hey userxyz you're code quality has been declining, you getting enough sleep?" 
B: "I'm narcoleptic you insensitive clod!"
A: "My bad"

Or even if they don't want to disclose the reason for the complaint:

A: "Hey userxyz you're code quality has been declining, you getting enough sleep?" 
B: "Please don't bring this up again"
A: "Bring what up?"
B: "Anything to do with sleep."
A: "Alright."

We've all had a moment where we though a caller on the phone was a different gender/child/etc, or congratulated someone on having a bun in the oven when they didn't. Or called a Sir, Ma'am. Or pulled a $20 out instead of your drivers license when pulled over (I wasn't thinking nor trying to bribe, I just knew it was going to cost me money and had a brain-fart. If he wasn't understanding I could have a criminal record for a very ridiculous reason.) 

This doesn't work for overt and obvious discrimination, but for anything that is in a grey area this seems important. 

> Prohibit vexatious and bad faith complaints.

:+1: 

Or repeated/excessive, I don't need anyone to protect me. I'm fine with someone submitting on behalf of a friend for certain situations, but if someones only "contribution" is complaining about things I can't see that resulting in a very productive community. 

This seems important, this is meant to protect people, not to be used as a weapon to hit others with, nor to enlist a PC militia to patrol our repos. 

One last thing, I'm not a fan of padding the CoC out with a big (and statedly not complete) list of things. Perhaps a few as an example, but making this an encyclopedia of "protected" things, makes it hard to read.

**Comment by ghost** [comment]
> Would a clause that gives an opportunity to have why this is harassment be explained make sense? and an opportunity to make amends/explain? aka "Never attribute to malice what can equally be explained by ignorance."

Note the - also self-serving - "unwillingness to discuss social justice issues" line, which answers your question. What some hardline SJ advocates would say now would be that you're a "concern-trolling misogynist". Enjoy!

**Comment by An-Annoymous-Developer** [comment]
> However, for protecting subcultures more generally I'm less sure, since subcultures are often proxies for political conflicts over values.

ISIS is a bad example since they're not a subculture, they're a formal organisation with leadership and centrally defined policy. It's ok to assume that a member of ISIS agrees with their policies and judge them for it. 

Anti-vaxxers should be a protected category. They're wrong, but they're not intentionally hurting anyone. The difference between anti-vax and Galileo's heretical "the Earth goes around the sun" is simply that Galileo was right and anti-vaxxers are wrong. But how will we ever discover who is right or wrong without those ideas and their supporters being aloud to discuss them in public?

If the CoC sets rules for when and how discussion on controversial topics is acceptable, it's not required to exclude members with controversial beliefs. 

Or to put it another way. Why is "my religion says vaccination is a sin" something the code of conduct explicitly protects and "I believe vaccination causes autism" something that's so severe it needs to be explicitly excluded from protection? 

And now we get to membership of hate groups. The question becomes what's a bigger risk: That without this clause someone might be victimised for membership in a harmless group (like goths) or that with it someone might not be excluded despite membership of the KKK. I think the far bigger risk is what happens without the clause because: 

1) Members of a hate group attending an open source conference would most likely be smart enough not to tell anyone they're members of a hate group. 
2) If they're not smart enough they'll most likely be easily excluded under another clause.

**Comment by KelseyDH** [comment]
It's worth pointing out that some forms of discrimination are _legitimate_.  Just because a class can be identifiable, that does not mean that the class is _necessarily_ worthy of _rights protection_ just because it can be identified.  Our society discriminates against people or groups _legitimately_ all the time.  If you don't show up for work you don't get a promotion.  If you harm others, you go to jail.  If you're under a certain age, you can't do certain things. If you aren't educated to be a doctor, you can't perform surgery -- All forms of discrimination, but legitimate nonetheless.  Meanwhile, discrimination on the basis of race, sex, religion, etc are prohibited, because as a society we recognize that discriminating on these grounds is (almost always) baseless, exclusionary, arbitrary, unfair, marginalizing and ultimately, _illegitimate._

---

**RE: 'unwelcomed comments' and 'bona fide justification'**

@atarzwell Honestly I agree with you.  Allowing a `bona fide justification` could allow pretty much everything.  It was my attempt at compromise.  The problem here is not with the language I added, but with the original statement:

```
Unwelcome comments without bona fide justification regarding a person’s lifestyle choices and practices, including those related to food, health, parenting, drugs, and employment
```

Trying to prohibit all `"unwelcome comments"` over all domains of life is realistically unworkable for a harassment policy.  It's far too broad, all encompassing, and lacking in nuance. The sentence should be removed in its entirety and replaced with something more general encouraging civility.

---

**RE: protecting Subcultures**

@An-Annoymous-Developer Criminal gangs are an identifable subculture in addition to being an institution.  People in gangs are free to generally express and wear clothing as they like, but it would be absurd to say that they deserve privileged rights protection as a _separately protected class._  If a local business refused to do business with a black person on the basis of their race, human rights laws kick in because society agrees that form of discrimination is illegitimate.  If subcultures were added as identifiable class, this expansive definition of protection could lead to many subcultures becoming entitled to explicit protection that could interfere with social rights more than it protects them.  A gang could demand the use of a church.  A techno club could be forced to play country music.  A sports team could be forced to say nice things about its rival.  The ramifications are endless, and we have to question what good such protection is really trying to achieve -- though I'm not fully ruling out supporting protections for subcultures just yet.

File: index.md [hunk 1]
```diff
@@ -34,21 +39,24 @@ Harassment includes, but is not limited to:
 - Deliberate “outing” of any aspect of a person’s identity without their consent except as necessary to protect others from intentional abuse
 - Publication of non-harassing private communication
 
-Our open source community prioritizes marginalized people’s safety over privileged people’s comfort. We will not act on complaints regarding:
+We will not act on complaints regarding:
 
-- ‘Reverse’ -isms, including ‘reverse racism,’ ‘reverse sexism,’ and ‘cisphobia’
+- Good faith and non-malicious conduct whose object is to ameliorate the conditions of disadvantaged individuals or groups including those that are disadvantaged because of race, national or ethnic origin, colour, religion, sex, gender identity, sexual orientation, physical appearance, body size, age or mental or physical disability.
```
Comment (Addvilz) [inline index.md:44]: I can not emphasize enough how much we need this change! :+1: Good work!

**Comment by Addvilz** [comment]
:+1:

File: index.md (hunk 1 above)
Comment (LimeBlast) [inline index.md:44]: :+1:

**Comment by bkeepers** [comment]
Thanks for your feedback. See https://github.com/todogroup/opencodeofconduct/issues/84.

**Pull request closed without merging**